pandas-stubs/core/strings.pyi

```diff
@@ -44,9 +52,19 @@ class StringMethods(NoNewAttributesMixin, Generic[T]):
         na_rep: str | None = ...,
         join: Literal["left", "right", "outer", "inner"] = ...,
     ) -> T: ...
+    @overload
+    def split(
+        self, pat: str = ..., n: int = ..., *, expand: Literal[True], regex: bool = ...
+    ) -> _TS: ...
```

Comment:
it might be nice to have all the `*` after the same argument and not at different levels across overloads.

Reply:
done in next commit